Update AWS Connection docs and deprecate some extras (#24670)
mistercrunch/Airflow: docs/apache-airflow-providers-amazon/connections/aws.rst

Extra (optional)
    Specify the extra parameters (as json dictionary) that can be used in AWS
    connection. The following parameters are all optional:

    * ``aws_access_key_id``: AWS access key ID used for the initial connection.
    * ``aws_secret_access_key``: AWS secret access key used for the initial connection
    * ``aws_session_token``: AWS session token used for the initial connection if you use external credentials.
      You are responsible for renewing these.
    * ``region_name``: AWS Region for the connection.
    * ``session_kwargs``: Additional **kwargs** passed to
      `boto3.session.Session <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html>`__.
    * ``config_kwargs``: Additional **kwargs** used to construct a
      `botocore.config.Config <https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html>`__
      passed to `boto3.client <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client>`__
      and `boto3.resource <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.resource>`__.

>     * ``role_arn``: If specified, then use this role for obtain a set of temporary security credentials will be done to this role.

Reviewer: ```suggestion
    * ``role_arn``: If specified, then assume this role, obtaining a set of temporary security credentials using the ``assume_role_method``.
```

    * ``assume_role_method``: `AWS STS Client method <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#client>`__
      for `assume role <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html>`__,

>       if not specified than `assume_role <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.assume_role>`__ use.

Reviewer: I find this section a bit confusing the way it's written now, maybe just list the acceptable values with a short description and link to the docs for each:
1. `assume_role_with_web_identity`  ([doc link](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html))
1. `assume_role`  ([doc link](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html))
1. `assume_role_with_saml` ([doc link](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithSAML.html))

Author: Another problem with assume_role, related to `with_saml` and `with_web_identity` at that moment.
Each of current implementation (one for saml one for web-identity) created additional parameters or use parameters in their body for send in assume_role_method.

I have a plan to move this outside of `BaseSessionFactory` and make more generic.

    * ``assume_role_kwargs``: Additional **kwargs** passed to ``assume_role_method``.
    * ``host``: Endpoint URL for the connection.
    * ``s3_config_file``: Path to local credentials file.

>     * ``s3_config_format``: ``s3_config_file`` format, one of

Reviewer: If I read your descriptions of this PR right, then I think we agree that these shouldn't be here. I wouldn't add documentation for it now if we're planning on removing it and don't want people to use it. At the very least, note here that these are deprecated.

Author: Yep, I will add into deprecation section and add warning for user who probably still use this type of authentication.

      `aws <https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-settings>`_,
      `boto <http://boto.cloudhackers.com/en/latest/boto_config_tut.html#details>`_ or

>       `s3cmd <https://s3tools.org/kb/item14.htm>`_ if not specified than **boto** use.

Reviewer: ```suggestion
      `s3cmd <https://s3tools.org/kb/item14.htm>`_ if not specified then **boto** is used.
```

    * ``profile``: If you are getting your credentials from the ``s3_config_file``
      you can specify the profile with this parameter.

> .. warning:: Extra parameters below deprecated and will remove in future version of provider.

Reviewer: ```suggestion
.. warning:: Extra parameters below are deprecated and will be removed in a future version of this provider.
```

Reviewer: Also I'd add the s3 related ones above down here.
